### ISSUE TITLE
[HOLD for payment 2023-05-03] [$1000] Long pressing any of the app download link on mweb Safari does not open the copy to clipboard menu

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#16946]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Login to any account
2. Open the App download links in the About section.
3. Long press on any of the link.

## Expected result:
Copy to clipboard menu item should show up

## Actual result:
Copy to clipboard menu item does not show up

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [x] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.2.94-0
**Reproducible in staging?:** y 
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/43996225/229945424-70ebe365-259b-475f-9996-608b37561013.MP4

https://user-images.githubusercontent.com/43996225/229945458-0bfbae2f-30d0-41ad-a957-955a409df300.MP4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @mallenexpensify 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1680564052842509

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~014fb329f5e82a7436</li>
        <li>Upwork Job ID: 1643618667751460864</li>
        <li>Last Price Increase: 2023-04-05</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @abekkala (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by koko57** [comment]
Hi I'm Agata from Callstack - expert contributor group - maybe we could help with this one?

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~014fb329f5e82a7436

**Comment by MelvinBot** [comment]
Current assignee @abekkala is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @eVoloshchak (`External`)

**Comment by MelvinBot** [comment]
Current assignee @mountiny is eligible for the External assigner, not assigning anyone new.

**Comment by akinwale** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
When the app download links are displayed, long-pressing one of the menu items should display the "Copy to clipboard" menu item expected in a custom context menu, but the menu item text is being highlighted instead.

### What is the root cause of that problem?
The root component of the custom `MenuItem` component is a React Native `Pressable`, which is capturing the `onPressIn` and `onPressOut` events, causing the event handlers for the similarly named events in the parent `PressableWithSecondaryInteraction` component to be ignored.

### What changes do you think we should make in order to solve the problem?
After some testing, it appears that `MenuItem` should be refactored to make use of a `PressableWithSecondaryInteraction` component as its root component. A `PressableWithSecondaryInteraction` is already based on the `Pressable` component, so this shouldn't be a breaking change.

However, this would require refactoring `PressableWithSecondaryInteraction` to set the `style` prop to make use of a `style` passed as a property if the `inline` prop is false, in order to maintain compatibility with other components or pages that are already making use of the `PressableWithSecondaryInteraction` component.

https://github.com/Expensify/App/blob/0d2d52eaa4f5ce91585be007645ad1d4e6a230df/src/components/PressableWithSecondaryInteraction/index.js#L57

The style property would be refactored like so:
`style={this.props.inline ? styles.dInline : this.props.style}`

`MenuItem` would also be refactored to accept the following props: `onPressIn`, `onPressOut` and `onSecondaryInteraction`.

### What alternative solutions did you explore?
Attempted to make use of the `stopPropagation()` and `event.nativeEvent.stopImmediatePropagation()` methods for the event parameter with the `onPressIn` and `onPressOut` event handlers, but these calls did not work.

**Comment by ImBIOS** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

When long-pressing on a menu item in the app download links, the custom context menu is not being displayed correctly. Instead of showing a "Copy to clipboard" option, the menu item text is being highlighted.

### What is the root cause of that problem?

The app download links are part of an existing codebase that uses React Native. The MenuItem component, which is used to display the menu items, has a root component of Pressable, which is causing the issue.

### What changes do you think we should make in order to solve the problem?


#### Proposed solution

Refactor the MenuItem component to use the PressableWithSecondaryInteraction component as its root component. The PressableWithSecondaryInteraction component is based on the Pressable component, but it also includes support for secondary interactions like long presses. By using this component as the root, we can ensure that the custom context menu is displayed correctly when long-pressing on a menu item.

#### Benefits
Refactoring the MenuItem component to use PressableWithSecondaryInteraction will provide the following benefits:

The custom context menu will be displayed correctly when long-pressing on a menu item.
The MenuItem component will be more consistent with other components in the app that use the PressableWithSecondaryInteraction component.
The solution should be relatively straightforward to implement and test.
Implementation steps
Modify the MenuItem component to use the PressableWithSecondaryInteraction component as its root component.
Add the onPressIn, onPressOut, and onSecondaryInteraction props to the MenuItem component.
Update the style prop in PressableWithSecondaryInteraction to make use of the inline prop if it is provided, or use the style prop if the inline is false.
Test the MenuItem component to ensure that the custom context menu is displayed correctly when long-pressing on a menu item.

#### Potential drawbacks

The proposed solution should not cause any significant performance issues or require a significant amount of development time. However, there may be some minor changes required in other parts of the app that use the MenuItem component, as they may need to be updated to include the new onPressIn, onPressOut, and onSecondaryInteraction props.

Overall, the proposed solution provides a straightforward and effective way to address the issue of the custom context menu not being displayed correctly.

**Comment by mountiny** [comment]
@eVoloshchak please review the proposals once you get time!

**Comment by MelvinBot** [comment]
@eVoloshchak, @abekkala, @mountiny Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by MelvinBot** [comment]
@eVoloshchak, @abekkala, @mountiny Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by eVoloshchak** [comment]
Not overdue, reviewing proposals today

**Comment by MelvinBot** [comment]
📣 @b1tjoy! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by MelvinBot** [comment]
✅ Contributor details stored successfully. Thank you for contributing to Expensify!

**Comment by eVoloshchak** [comment]
@akinwale's [proposal](https://github.com/Expensify/App/issues/16946#issuecomment-1498083798) is correct in identifying the root cause. 
I think removing the PressableWithSecondaryInteraction [here](https://github.com/Expensify/App/blob/main/src/pages/settings/AppDownloadLinks.js#L84) and refactoring MenuItem to use PressableWithSecondaryInteraction is a good approach, since it allows us to get rid of nested pressables. 
However, it adds a bug in Safari, the whole screen is highlighted when you long press


https://user-images.githubusercontent.com/9059945/231171261-6efc4d84-0028-42e8-99b1-76e6c5b0b5d3.MP4


@ImBIOS, correct me if I'm wrong, but your [proposal](https://github.com/Expensify/App/issues/16946#issuecomment-1498323700) uses approach identical to @akinwale's proposal

**Comment by akinwale** [comment]
> However, it adds a bug in Safari, the whole screen is highlighted when you long press

@eVoloshchak This is not a bug. This is intended behaviour. The idea is to have a modal with a primary action displayed at the bottom. This primary action is what actually copies the link. You can test this in Android Chrome to confirm.

**Comment by eVoloshchak** [comment]
> @eVoloshchak This is not a bug. This is intended behaviour. The idea is to have a modal with a primary action displayed at the bottom. This primary action is what actually copies the link. You can test this in Android Chrome to confirm.

I don't see the same issue on Android Chrome


https://user-images.githubusercontent.com/9059945/231173542-45a678d2-d9d3-4efa-b21c-9e1273f831fd.mp4

**Comment by akinwale** [comment]
> > @eVoloshchak This is not a bug. This is intended behaviour. The idea is to have a modal with a primary action displayed at the bottom. This primary action is what actually copies the link. You can test this in Android Chrome to confirm.
> 
> I don't see the same issue on Android Chrome

Unless I'm missing something, I'm seeing the same behaviour. The modal background just appears to be darker in the iOS video. Or do you mean that the whole screen becomes highlighted before the long press effect is completed? I'm not seeing that as a bug, but I can find a solution for that.

**Comment by eVoloshchak** [comment]
@akinwale, notice the blue border around the screen on IOS

**Comment by akinwale** [comment]
> @akinwale, notice the blue border around the screen on IOS

Oh, I actually didn't notice that at all. Nothing fundamental has changed with the implementation, we're just changing where the events are being handled, so I'm guessing it's related to the selection feature.

**Comment by mountiny** [comment]
@eVoloshchak Yes I think any menu item which is a link should have this behaviour.

**Comment by mountiny** [comment]
Any good proposals or could we give this to Callstack to investigate? @eVoloshchak

**Comment by eVoloshchak** [comment]
@mountiny, there are a couple of good proposals, but both create one small issue that needs to be resolved. I think we might wait a little for @akinwale or @b1tjoy to update their proposals

**Comment by akinwale** [comment]
> > @b1tjoy, good job on finding the offending PR, and thanks for the thorough [explanation](https://github.com/Expensify/App/issues/16946#issuecomment-1501861485)
> > I can still see the bug with screen being highlighted with blue border in Safari, but weirdly, only with a physical device, the simulator doesn't have this issue. ControlSelection.block() in onPressIn handler and ControlSelection.unblock() in onPressOut handler are present, but I can still reproduce this on a real device
> 
> Ah, that explains some of it. I am also not seeing it in the iOS simulator. I will try to test with a real device to see if I can reproduce.

@eVoloshchak I'm not seeing the blue outline on an iPhone 12 Pro Max running iOS 16.3.1. Video attached. What iOS version are you running?

For reference, this is what the code listing would look like for the affected pages / components.
*AppDownloadLinks.js*
Remove `PressableWithSecondaryInteraction` from the map and use the `MenuItem` as the top level element.
```
<MenuItem
    title={props.translate(item.translationKey)}
    icon={item.icon}
    iconRight={item.iconRight}
    onPress={() => item.action()}
    shouldShowRightIcon
    key={item.translationKey}
    onPressIn={() => props.isSmallScreenWidth && DeviceCapabilities.canUseTouchScreen() && ControlSelection.block()}
    onPressOut={() => ControlSelection.unblock()}
    onSecondaryInteraction={e => showPopover(e, item.link)}
    ref={el => popoverAnchor = el}
    onKeyDown={(event) => {
        event.target.blur();
    }}
/>
```

*MenuItem.js*
```
<PressableWithSecondaryInteraction
    onPress={(e) => {
        if (props.disabled) {
            return;
        }

        if (e && e.type === 'click') {
            e.currentTarget.blur();
        }

        props.onPress(e);
    }}
    onPressIn={props.onPressIn}
    onPressOut={props.onPressOut}
    onSecondaryInteraction={props.onSecondaryInteraction}
    // ...
</PressableWithSecondaryInteraction>
```

*PressableWithSecondaryInteraction*
```
<Pressable
    style={this.props.inline ? styles.dInline : this.props.style}
    // ...
```

https://user-images.githubusercontent.com/4160319/231193287-0be26e0a-3559-4206-8f58-65f6ff9f2445.mp4

**Comment by akinwale** [comment]
> > @b1tjoy, good job on finding the offending PR, and thanks for the thorough [explanation](https://github.com/Expensify/App/issues/16946#issuecomment-1501861485)
> > I can still see the bug with screen being highlighted with blue border in Safari, but weirdly, only with a physical device, the simulator doesn't have this issue. ControlSelection.block() in onPressIn handler and ControlSelection.unblock() in onPressOut handler are present, but I can still reproduce this on a real device
> 
> Ah, that explains some of it. I am also not seeing it in the iOS simulator. I will try to test with a real device to see if I can reproduce.

@eVoloshchak I'm not seeing the blue outline on an iPhone 12 Pro Max running iOS 16.3.1. Video attached. What iOS version are you running?

For reference, this is what the code listing would look like for the affected pages / components.
*AppDownloadLinks.js*
Remove `PressableWithSecondaryInteraction` from the map and use the `MenuItem` as the top level element.
```
<MenuItem
    title={props.translate(item.translationKey)}
    icon={item.icon}
    iconRight={item.iconRight}
    onPress={() => item.action()}
    shouldShowRightIcon
    key={item.translationKey}
    onPressIn={() => props.isSmallScreenWidth && DeviceCapabilities.canUseTouchScreen() && ControlSelection.block()}
    onPressOut={() => ControlSelection.unblock()}
    onSecondaryInteraction={e => showPopover(e, item.link)}
    ref={el => popoverAnchor = el}
    onKeyDown={(event) => {
        event.target.blur();
    }}
/>
```

*MenuItem.js*
```
<PressableWithSecondaryInteraction
    onPress={(e) => {
        if (props.disabled) {
            return;
        }

        if (e && e.type === 'click') {
            e.currentTarget.blur();
        }

        props.onPress(e);
    }}
    onPressIn={props.onPressIn}
    onPressOut={props.onPressOut}
    onSecondaryInteraction={props.onSecondaryInteraction}
    // ...
</PressableWithSecondaryInteraction>
```

*PressableWithSecondaryInteraction*
```
<Pressable
    style={this.props.inline ? styles.dInline : this.props.style}
    // ...
```

https://user-images.githubusercontent.com/4160319/231193287-0be26e0a-3559-4206-8f58-65f6ff9f2445.mp4

**Comment by eVoloshchak** [comment]
@akinwale, I can still reproduce this using the code you've provided. 
I've tested on iPhone SE running iOS 16.3.1

**Comment by akinwale** [comment]
> @akinwale, I can still reproduce this using the code you've provided. I've tested on iPhone SE running iOS 16.3.1

Weird. I've also tested on a 2020 iPhone SE running iOS 16.3.1 and I still don't see the behaviour. Can you check if you have Full Keyboard Access in Settings > Accessibility > Keyboards enabled? If you do, can you turn it off, restart your device and then test again? Thanks.

**Comment by akinwale** [comment]
@eVoloshchak
Alternatively, could you try adding this CSS to `index.js` to see if it changes anything?
```
*:focus:not(.focus-visible) {
  outline: none;
}
```

**Comment by eVoloshchak** [comment]
> @eVoloshchak Can you reproduce whole page highlight on Safari using this proposal [#16946 (comment)](https://github.com/Expensify/App/issues/16946#issuecomment-1501861485)? I test it on real device but works fine.

Yes, I was also able to reproduce this with your proposal

**Comment by eVoloshchak** [comment]
@akinwale, huh, I think I've found why we have different results. 
 The issue is present only if app links page was the first page you entered (if you navigated to `settings/about`), if you enter this page from another screen - it's not present. 
 Notice in the video below the issue starts appearing only after I refresh the page
 
 

https://user-images.githubusercontent.com/9059945/231214014-596a475d-7cae-4013-98c7-ed93cc47f3b2.MP4

**Comment by akinwale** [comment]
@eVoloshchak This is the CSS pseudo selector affecting the output.
https://github.com/Expensify/App/blob/99292fc00d0c12817b5dde204156e6d100b662ec/web/index.html#L45-L48

I'm not sure what exactly it's meant for, but after commenting it out, the blue border doesn't show up any more upon refresh. We need to find a way to make the selector more restrictive.

**Comment by eVoloshchak** [comment]
I can also reproduce this on other screens, so this is a separate issue, I'll report it on slack. 

Could you post an updated proposal so as per [this](https://github.com/Expensify/App/issues/16946#issuecomment-1503406840) comment 
> any menu item which is a link should have this behaviour.

**Comment by akinwale** [comment]
> I can also reproduce this on other screens, so this is a separate issue, I'll report it on slack.
> 
> Could you post an updated proposal so as per [this](https://github.com/Expensify/App/issues/16946#issuecomment-1503406840) comment
> 
> > any menu item which is a link should have this behaviour.

Done.

**Comment by akinwale** [comment]
## Proposal
[Updated](https://github.com/Expensify/App/issues/16946#issuecomment-1498083798)

**Comment by MelvinBot** [comment]
📣 @akinwale You have been assigned to this job by @mountiny!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by mountiny** [comment]
Thanks, assigned!

**Comment by akinwale** [comment]
Applied to the Upwork job. Should have a PR ready within an hour.

**Comment by mountiny** [comment]
@abekkala Noting this is eligible for the speed bonus.

**Comment by akinwale** [comment]
@eVoloshchak Before this gets closed out, I just wanted to say thanks for the thorough review. As a relatively new contributor, I don't think I would've been able to independently identify the vibration bug during testing. It's been a great learning experience and I hope to gain more insight as I take on more tasks over time.

**Comment by eVoloshchak** [comment]
@akinwale, thanks for your message and for your kind words!
You did a great job on the code, and your fast response time made it easy for us to complete the task quickly.
It's awesome to have new contributors, looking forward to working with you on more tasks in the future.

**Comment by MelvinBot** [comment]
⚠️ Looks like this issue was linked to a **Deploy Blocker** [here](https://github.com/Expensify/App/issues/17580)

If you are the assigned CME please investigate whether the linked PR caused a regression and leave a comment with the results.

If a regression has occurred and you are the assigned CM follow the instructions [here](https://stackoverflow.com/c/expensify/questions/13647).

If this regression could have been avoided please consider also proposing a recommendation to the PR checklist so that we can avoid it in the future.

**Comment by mountiny** [comment]
PR was reverted and new PR is up

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by abekkala** [comment]
**Payments to be made on 2023-04-26**
Issue reported by: @mallenexpensify - no payout for you. 😆 
Selected Proposal for fix: @akinwale  [$1000] + [$500 PR Bonus]
C+ Review: @eVoloshchak [$1000] + [$500 PR Bonus]

Timeline:
Apr 11 - @akinwale was assigned to this job
Apr 11 - PR merged #17310 - 50% BONUS 🎉

**Comment by abekkala** [comment]
@akinwale and @eVoloshchak 
I have hired you both in Upwork and sent contracts. The $500 bonus will be added when I pay out! 🎉

**Comment by akinwale** [comment]
> @akinwale and @eVoloshchak I have hired you both in Upwork and sent contracts. The $500 bonus will be added when I pay out on the 26th! 🎉

Accepted. Thanks!

**Comment by mountiny** [comment]
@abekkala I think we need to slow down here, the PR was reverted so 26th is not correct date and technically it was regression, although hard to test for. 

I have updated the labels, lets get the new PR merged soon, I think we dont have to cut for the regression here given how active and helpful @akinwale ahas been here and it was a test case we did not anticipate.

But we should wait for the payout once the PR actually hits production (not even at staging now).

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.3.2-5 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/17310


If no regressions arise, payment will be issued on 2023-04-27. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@eVoloshchak] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@eVoloshchak] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@eVoloshchak] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@abekkala] Determine if we should create a regression test for this bug.
- [ ] [@eVoloshchak] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@abekkala] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by abekkala** [comment]
@vitHoracek 

> But we should wait for the payout once the PR actually hits production (not even at staging now)

Correct, I don't make any payments until I know everything is done/correct! I check everything on the 27th to ensure everything has been done before sending any payments out

.

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.3.5-6 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/17639


If no regressions arise, payment will be issued on 2023-05-03. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@eVoloshchak] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@eVoloshchak] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@eVoloshchak] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@eVoloshchak] Determine if we should create a regression test for this bug.
- [ ] [@eVoloshchak] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@abekkala] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by abekkala** [comment]
@mountiny I'm not sure why the title updated the payment date twice but didn't `replace` the date.
Can the 4-27 date be removed and payout is actually 05-03 now?

**Comment by mountiny** [comment]
>  I'm not sure why the title updated the payment date twice but didn't replace the date. Can the 4-27 date be removed and payout is actually 05-03 now?

@abekkala correct always consider the later one unless said otherwise

But the code is dumb and adds the new hold without looking if there is already one 😄

**Comment by eVoloshchak** [comment]
Not overdue, will fill out first 3 items of the BZ checklist today

**Comment by mountiny** [comment]
@eVoloshchak Bump

**Comment by eVoloshchak** [comment]
- The PR that introduced the bug has been identified. Link to the PR: 
n/a, this was missed during the initial implementation. As @b1tjoy said [here](https://github.com/Expensify/App/issues/16946#issuecomment-1501861485),
This was caused by https://github.com/Expensify/App/pull/12987 for AboutPage specifically, but we ended up adding link context menus to all of the menu items on all pages, which was missed during the initial implementation
- The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: n/a
- A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: This should be caught during the PR review, I think there isn't much more we can do. We can add a checklist item `All of the menu items should have a link context menu`, but that seems obvious. 

- Determine if we should create a regression test for this bug.
    >Is it easy to test for this bug?

    Yes
    > Is the bug related to an important user flow? (For example, adding a bank account)

    No
    > Is it an impactful bug?

    No
    
    This is a non-breaking bug, that isn't affecting user flow. I don't think the regression test is needed here. It would mean testing that every menu item that has a link also has a context menu with the said link. A simple 1 step test, but for every menu item in the app.

**Comment by abekkala** [comment]
@eVoloshchak and @akinwale 
You've each been paid and contract ended in Upwork 🎉 
Thank you!

**Comment by eVoloshchak** [comment]
@abekkala, I think the amount should be $500 instead of 1500$
This has caused a regression

**Comment by eVoloshchak** [comment]
@abekkala, bump on the above

**Comment by mountiny** [comment]
Not sure whats the process on this one, maybe we can just cut the price down on some future payments

**Comment by abekkala** [comment]
hmm, yeah, I can't retract a payment.

> Each regression reduces the pay amount for C+ by 50% (only C+ gets reduced pay). Additionally, no efficiency bonus is paid for merging quickly if there are any regressions.

So yes, you're correct your payment for this should have been only $500 vs $1500. 
A future job of yours will need to be reduced by $1000 to make up for the overpayment here.

**Comment by mountiny** [comment]
@akinwale can you link some issue where you active that we could reduce the payment for to compensate for this please?

**Comment by mountiny** [comment]
Oh damn yeah, was there no payment on the follow up issues/regression?

**Comment by akinwale** [comment]
> Oh damn yeah, was there no payment on the follow up issues/regression?

No, there wasn't. @eVoloshchak did the review for the follow up issue/regression as well (https://github.com/Expensify/App/issues/17656#issuecomment-1515046396). Note that this also covered a new issue that was discovered in a different commit.

**Comment by mountiny** [comment]
@abekkala ok so we are all good here, thanks for raising this. With millions of issues its easy to get lost in this

**Comment by abekkala** [comment]
@mountiny Only the C+ gets a reduced payment in the case of a regression. (in this case it would have meant a reduction for evoloshchak only - not akinwale)
Are you saying that there actually isn't a need for a reduction in payment due to this being an edge case? 

I have a payment coming up for evoloshchak (on a different GH)  that I was going to reduce in order to even out this GH. Is that not necessary and I can just do a normal full pay out without reducing due to this GH regressIon?

**Comment by mountiny** [comment]
Yes full pay out!

Oh I thought its both who get reduced payment.

**Comment by mountiny** [comment]
I think everything was done and paid here so closing, thanks for all the help, please comment if otherwise
